File: Ilxomjonov_Olimjon/core/factories/subsystem_factory.dart
import '../../modules/energy/energy.dart';
import '../../modules/lighting/lighting.dart';
import '../../modules/security/security.dart';
import '../../modules/transport/transport.dart';

abstract class SubsystemFactory {
  dynamic createSubSysmte();
}

class LightingFactory implements SubsystemFactory {
  @override
  Lighting createSubSysmte() => Lighting();
}

class TransportFactory implements SubsystemFactory {
  @override
  Transport createSubSysmte() => Transport();
}

class SecurityFactory implements SubsystemFactory {
  @override
  Security createSubSysmte() => Security();
}


class EnergyFactory implements SubsystemFactory{
  @override
  Energy createSubSysmte() => Energy();
}

File: maxsudaliyev-mirzohidjon/core/factories/SubsystemFactory.php
<?php
namespace core\factories;

use modules\lighting\LightingSystem;
use modules\transport\TransportSystem;
use modules\security\SecuritySystem;
use modules\energy\EnergySystem;
use core\adapters\WeatherService;

/**
 * PATTERN: Abstract Factory
 * Purpose: Create related subsystem objects consistently (lighting/transport/security/energy + integrations).
 */
interface SubsystemFactory
{
    public function createLighting(): LightingSystem;
    public function createTransport(): TransportSystem;
    public function createSecurity(): SecuritySystem;
    public function createEnergy(): EnergySystem;
    public function createWeatherService(): WeatherService;
}
